File: pset6/greedy.py
#!/usr/bin/env python3

import cs50

def main():
    print("O hai! how much change is owned? ")
    while True:
            change = cs50.get_float()
            if change > 0:
                break
    
    cents = int(round(change* 100))
    
    counter = 0
    
    counter += cents // 25
    cents %= 25
  
    counter += cents // 10
    cents %= 10
    
    counter += cents // 5
    cents %= 5
    
    counter += cents
    
    print(counter)
    
        
    

if __name__ == "__main__":
    main()

File: pset1/Greedy/greedy.c
#include<cs50.h>
#include<stdio.h>
#include<math.h>
float GetPositiveFloat();

int main(void)
{
    int quaters = 25, dimes = 10, nickels = 5, pennies = 1,counter = 0;
    printf("O hai! "); 
    
    //roundup
    int in_cents = roundf((GetPositiveFloat()*100)) ;
    
    while(in_cents>= quaters)
    {
        in_cents -= quaters;
        counter++;
        
    }
   
   while(in_cents>= dimes)
   {
        in_cents -= dimes;
        counter++;
    }
    
    while(in_cents>= nickels)
    {
        in_cents -= nickels;
        counter++;
    }
    
    while(in_cents>=pennies)
    {
        in_cents -= pennies;
        counter++;
    }
    
    printf("%d\n",counter);
    
}

float GetPositiveFloat(void)
{
    float n;
    do
    {
        printf("How much change is owed?\n");
        n = GetFloat();
    }while(n<0);
    
   return n;
}